src/libutil/filesystem.cpp

```diff
@@ -1013,4 +1013,129 @@ Filesystem::scan_for_matching_filenames(const std::string &pattern_,
     return true;
 }
 
+
+
+Filesystem::IOFile::IOFile (string_view filename, Mode mode)
+    : IOProxy(filename,mode)
+{
+    // Call Filesystem::fopen since it handles UTF-8 file paths on Windows,
+    // which std fopen does not.
+    m_file = Filesystem::fopen (m_filename.c_str(), mode == Write ? "wb" : "rb");
+    if (! m_file)
+        mode = Closed;
+    m_auto_close = true;
```

Comment:
Why do you need this flag?

Reply:
Because if you construct IOFile with a filename, it opens and owns (and needs to close) the FILE*.

But if you construct IOFile with an open FILE*, it's owned by the caller and the IOFile should not close upon destruction.

Reply:
Oh I missed that other constructor. Makes sense.